docs/pages/example/mapbox-gl-geocoder-limit-region.html

```diff
@@ -0,0 +1,34 @@
+<script src='https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-geocoder/v2.2.0/mapbox-gl-geocoder.min.js'></script>
+<link rel='stylesheet' href='https://api.mapbox.com/mapbox-gl-js/plugins/mapbox-gl-geocoder/v2.2.0/mapbox-gl-geocoder.css' type='text/css' />
+<div id='map'></div>
+
+<script>
+var map = new mapboxgl.Map({
+    container: 'map',
+    style: 'mapbox://styles/mapbox/streets-v9',
+    center: [146.359, -32.648],
+    zoom: 4
+});
+
+map.addControl(new MapboxGeocoder({
+    accessToken: mapboxgl.accessToken,
+
+    // limit results to Australia
+    country: 'au',
+
+    // further limit results to the geographic bounds representing the region of
+    // New South Wales
+    bbox: [139.965, -38.030, 155.258, -27.839],
+
+    // apply a client side filter to further limit results to those strictly within
+    // the New South Wales region
+    filter: function (item) {
+        // returns true if item contains New South Wales region
+        return item.context.map((i) => {
+            return (i.id.startsWith('region') && i.text === "New South Wales");
```

Comment:
[That's right](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith).

I've included a polyfill since I feel startsWith is much more readable than the alternative of `i.id.substr(0, 6) === 'region'` and simpler than a regexp.

Reply:
👍 looks like that just needs a `no-extend-native` line for tests to pass.

Reply:
Sorry, but after thinking some more I think it's probably better to simply split and shift since "id of the feature of the form {index}.{id}"